nautobot/extras/views.py

```diff
@@ -1559,7 +1559,12 @@ class JobLogEntryTableView(View):
 
     def get(self, request, pk=None):
         instance = self.queryset.get(pk=pk)
-        log_table = tables.JobLogEntryTable(data=instance.job_log_entries.all(), user=request.user)
+        filter_q = request.GET.get("q")
+        if filter_q:
+            queryset = instance.job_log_entries.filter(Q(message__iregex=filter_q) | Q(log_level__iregex=filter_q))
```

Comment:
I'm concerned about the potential for a ReDoS now that we're doing the filtering server-side rather than client-side. I wonder if it would be acceptable to reduce this down to simple `icontains` checks instead of `iregex`?

Reply:
I have the same concern but that would be a loss of functionality. Maybe we could add a delay so it's not sending requests on every single keystroke?

Reply:
I'm OK with losing functionality here personally.

Reply:
I'm concerned that users might rely on this heavily and this will negatively impact one of our core automation features.

Reply:
@lampwins

Reply:
As discussed, since this previously only worked on the first page of results it wasn't that useful anyway so we're going to change to icontains.